sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -1,255 +1,255 @@
+                methodSymbol.Parameters[0].Type.Is(KnownType.System_Runtime_Serialization_SerializationInfo) &&
+                methodSymbol.Parameters[1].Type.Is(KnownType.System_Runtime_Serialization_StreamingContext);
+        }
+
+        public static bool IsArrayClone(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.Ordinary &&
+                methodSymbol.Parameters.Length == 0 &&
+                methodSymbol.Name == nameof(Array.Clone) &&
+                methodSymbol.ContainingType.Is(KnownType.System_Array);
+        }
+
+        public static bool IsGcSuppressFinalize(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.Parameters.Length == 1 &&
+                methodSymbol.Name == nameof(GC.SuppressFinalize) &&
+                methodSymbol.ContainingType.Is(KnownType.System_GC);
+        }
+
+        public static bool IsDebugAssert(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.Name == nameof(Debug.Assert) &&
+                methodSymbol.ContainingType.Is(KnownType.System_Diagnostics_Debug);
+        }
+
+        public static bool IsOperatorBinaryPlus(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.UserDefinedOperator &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Name == "op_Addition";
+        }
+
+        public static bool IsOperatorBinaryMinus(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.UserDefinedOperator &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Name == "op_Subtraction";
+        }
+
+        public static bool IsOperatorEquals(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.UserDefinedOperator &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Name == "op_Equality";
+        }
+
+        public static bool IsOperatorNotEquals(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.UserDefinedOperator &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Name == "op_Inequality";
+        }
+
+        public static bool IsConsoleWriteLine(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.IsInType(KnownType.System_Console) &&
+                methodSymbol.Name == nameof(Console.WriteLine);
+        }
+
+        public static bool IsConsoleWrite(this IMethodSymbol methodSymbol)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'Microsoft.CodeAnalysis.ISymbol' instead of 'Microsoft.CodeAnalysis.IMethodSymbol'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)